src/DeepL.php

```diff
@@ -281,194 +342,34 @@ protected function checkLanguages($sourceLanguage, $destinationLanguage)
     }
 
     /**
-     * Build the URL for the DeepL API request
-     *
-     * @param string     $sourceLanguage
-     * @param string     $destinationLanguage
-     * @param string     $tagHandling
-     * @param array|null $ignoreTags
-     * @param string     $formality
-     * @param string     $resource
-     * @param null       $splitSentences
-     * @param null       $preserveFormatting
-     * @param array|null $nonSplittingTags
-     * @param null       $outlineDetection
-     * @param array|null $splittingTags
+     * @param array $paramsArray
      *
      * @return string
-     *
-     * @SuppressWarnings(PHPMD.NPathComplexity)
-     * @SuppressWarnings(PHPMD.CyclomaticComplexity)
-     * @SuppressWarnings(PHPMD.ExcessiveParameterList)
      */
-    protected function buildUrl(
-        $sourceLanguage = null,
-        $destinationLanguage = null,
-        $tagHandling = null,
-        array $ignoreTags = null,
-        $formality = 'default',
-        $resource = self::API_URL_RESOURCE_TRANSLATE,
-        $splitSentences = null,
-        $preserveFormatting = null,
-        array $nonSplittingTags = null,
-        $outlineDetection = 1,
-        array $splittingTags = null
+    protected function buildQuery(
+        $paramsArray
```

Comment:
code style: line breaks aren't necessary anymore

Reply:
cleaned it up